tests/cpp/data/test_sparse_page_dmatrix.cu

```diff
@@ -0,0 +1,26 @@
+// Copyright by Contributors
+
+#include <dmlc/filesystem.h>
+#include "../helpers.h"
+
+namespace xgboost {
+
+TEST(GPUSparsePageDMatrix, EllpackPage) {
+  dmlc::TemporaryDirectory tempdir;
+  const std::string tmp_file = tempdir.path + "/simple.libsvm";
+  CreateSimpleTestData(tmp_file);
+  DMatrix* dmat = DMatrix::Load(tmp_file + "#" + tmp_file + ".cache", true, false);
+
+  // Loop over the batches and assert the data is as expected
+  for (const auto& batch : dmat->GetBatches<EllpackPage>({0, 256, 64})) {
```

Comment:
Sorry for the nitpick, what if the parameter is changed, it might happen between runs.

Reply:
Good catch. I saved `BatchParam` and added some checks.